dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -171,6 +172,19 @@ default String getCleanDAG(String mainDescriptorPath, String mainDescriptor, Set
         return DAGHelper.cleanDAG(getContent(mainDescriptorPath, mainDescriptor, secondarySourceFiles, type, dao));
     }
 
+    default ParsedInformation getParsedInformation(Version version, DescriptorLanguage descriptorLanguage) {
+        Optional<ParsedInformation> foundParsedInformation = version.getVersionMetadata().getParsedInformationSet().stream()
+                .filter(parsedInformation -> parsedInformation.getDescriptorLanguage() == descriptorLanguage).findFirst();
+        if (foundParsedInformation.isPresent()) {
+            return foundParsedInformation.get();
+        } else {
+            ParsedInformation parsedInformation = new ParsedInformation();
+            parsedInformation.setDescriptorLanguage(descriptorLanguage);
+            version.getVersionMetadata().getParsedInformationSet().add(parsedInformation);
```

Comment:
This would normally result in a database record being persisted. However, if this ends up getting called from a read-only transaction, it won't be saved.

1. Maybe this doesn't get called from a read-only transaction (don't have the whole flow in my head), in which case this whole comment is moot.
2. It looks very lightweight, so no big deal if it's not saved?

Pointing it out just in case.

Reply:
- It's lightweight enough that's it's no big deal
- The language parsing code (the majority of CWLHandler and WDLHandler) would then have to be disabled alongside this which is outside the scope of this PR